Task 8.1

ArsenijDKS opened the pull request.

ArsenijDKS committed:
new file:   lib/main.dart

ArsenijDKS committed:
Merge remote-tracking branch 'origin/main' into task-8.1

ArsenijDKS committed:
modified:   README.md
	modified:   task-1/pubspec.lock

DmitrDomrachev commented on lib/main.dart:
```diff
@@ -0,0 +1,136 @@
+  int incrementCount = 0;
+  int dicrementCount = 0;
+  bool _hasError = false;
+  final _textEditingController = TextEditingController();
```
Эта штука нигде не используется. Не понятно, зачем ее добавил

DmitrDomrachev commented on lib/main.dart:
```diff
@@ -0,0 +1,136 @@
+
+class _MyHomePageState extends State<MyHomePage> {
+  int _counter = 0;
+  int incrementCount = 0;
+  int dicrementCount = 0;
```
Переменные нужно делать анонимными, если они не используются вне класса. Тип переменных обычно явно не указывают, если им сразу присваивает значение. Дают сам определит тип. `var _incrementCount = 0;`

DmitrDomrachev commented on lib/main.dart:
```diff
@@ -0,0 +1,136 @@
+      dicrementCount++;
+      _counter--;
+      if (_counter < 0) {
+        _hasError = true;
+      } else {
+        _hasError = false;
```
Тут что-то намудрил. не уверен, что нужно вообще было добавлять отдельную переменную для отображения ошибок.
Самый простой вариант - в начале метода проверять что counter > 0, и только тогда уменьшать его значение. В противном случае - ничего не делать
Если хочется отобразить, то я бы предложил использовать [cнэк](https://docs.flutter.dev/cookbook/design/snackbars) 
Если хочешь оставить свой вариант, то его необходимо доработать. Если я сейчас уменьшу счетчик до -1, а потом увеличу его значение до положительного - ошибка не сбросится.

DmitrDomrachev reviewed (commented):
Поправь по комментариям выше и присылай на проверку

ArsenijDKS committed:
modified:   lib/main.dart
    fixed - below zero bug

ArsenijDKS committed:
modified:   lib/main.dart

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+  var _incrementCount = 0;
+  var _dicrementCount = 0;
+  bool _hasError = false;
+  final _textEditingController = TextEditingController();
```
@ArsenijDKS Зачем вы сюда добавили `TextEditingController`? Он используется для полей ввода. У вас нигде нет таких полей, поэтому надо удалить.

При написании кода обращайте внимание на анализатор кода - он всегда посказывает что-то неправильно, что-то не используется и т.п.

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+
+  void _incrementCounter() {
+    if (_counter >= 0) {
+      _hasError = false;
```
Давайте пока не будем усложнять. Удалите пожалуйста проверку и ошибку. Оставьте только увеличение счетчика и подсчёт количества нажатий.

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+  }
+
+  void _dicrementCounter() {
+    if (_counter > 0) {
```
Давайте в этом методе сделаем проверку:
Eсли текущее значение 0, то уменьшить мы его не можем и покажем снек:
```dart
const snackBar = SnackBar(content: Text('Больше вы не можете уменьшить счётчик!'));
                ScaffoldMessenger.of(context).showSnackBar(snackBar);
                
``` 
Иначе уменьшим счётчик и увеличим количество нажатий на кнопку уменьшения

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+  void initState() {
+    super.initState();
+
+    _textEditingController.addListener(_dicrementCounter);
```
Давайте удалим - вы ведь это не используете 🤗

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+  }
+
+  @override
+  void dispose() {
```
Диспоз теперь тоже не нужен

internetova commented on lib/main.dart:
```diff
@@ -0,0 +1,143 @@
+              SizedBox(
+                height: 10,
+              ),
+              Text(_hasError
```
Это давайте удалим, будем показывать снек - выше я написала как.

internetova reviewed (commented):
Всё не так плохо. Давайте только немного улучшим и удалим лишнее.

Посмотрите пожалуйста внимательно мои комментарии и постарайтесь исправить по ним.

Ещё было бы замечательно если бы вы приложили видео как работает ваше приложение. Сжать размер видео файла можно в сервисе: https://clideo.com/ru/compress-video

ArsenijDKS committed:
modified:   lib/main.dart
    added: SnackBar
    removed: TextEditingController, Error check, dispose, Text(_hasError)

internetova commented on lib/main.dart:
```diff
@@ -34,45 +34,42 @@ class _MyHomePageState extends State<MyHomePage> {
   var _dicrementCount = 0;
   bool _hasError = false;
-  final _textEditingController = TextEditingController();
+  // final _textEditingController = TextEditingController();
```
удалим - это значит удалим

internetova commented on lib/main.dart:
```diff
@@ -34,45 +34,42 @@ class _MyHomePageState extends State<MyHomePage> {
-      });
-    }
+      }
+      if (_counter == 0) {
```
лучше использовать конструкцию `else`: если больше 0 - уменьшаем, иначе показываем снек

internetova reviewed (approved):
Зачем вы оставляете закомментрированный код?
Если оставляете, то следует писать TODO зачем тут этот код.

Так же я просила прикладывать демонстрацию как работает ваш код 🤗
